BACKWARDS_COMPATABILITY.md

```diff
@@ -0,0 +1,116 @@
+     - [Actions](https://docs.pepr.dev/main/user-guide/actions/)
+     - [Capabilities](https://docs.pepr.dev/main/user-guide/capabilities/)
+     - [Store](https://docs.pepr.dev/main/user-guide/store/)
+     - [OnSchedule](https://docs.pepr.dev/main/user-guide/onschedule/)
+1. **Configuration**:
+   - YAML or JSON configuration structures for [Pepr Modules](https://docs.pepr.dev/main/user-guide/pepr-modules/). Configuration includes:
+     - Environment Variables
+       - [Redact Store Values from Logs](https://docs.pepr.dev/main/user-guide/customization/#redact-store-values-from-logs)
+       - [Display Node Warnings](https://docs.pepr.dev/main/user-guide/customization/#display-node-warnings)
+       - [Customizing Log Format](https://docs.pepr.dev/main/user-guide/customization/#customizing-log-format)
+       - [Customizing Watch Configuration](https://docs.pepr.dev/main/user-guide/customization/#customizing-watch-configuration)
+       - [Configuring Reconcile](https://docs.pepr.dev/main/user-guide/customization/#configuring-reconcile)
+       - [Customizing with Helm](https://docs.pepr.dev/main/user-guide/customization/#customizing-with-helm)
+       - [Customizing with package.json](https://docs.pepr.dev/main/user-guide/customization/#customizing-with-packagejson)
+1. **Pepr SDK**:
+   - Helper functions to frequently used tasks defined in [SDK](https://docs.pepr.dev/main/user-guide/sdk/).
+1. **CLI Commands**:
+   - Commands, flags, and outputs provided by the [Pepr CLI](https://docs.pepr.dev/main/user-guide/pepr-cli/).
+
+---
+
+## Backward Compatibility Guarantees
+
+Pepr adheres to the following guarantees to ensure a seamless user experience:
+
+### 1. Semantic Versioning (SemVer)
+Pepr follows SemVer conventions:
+- **Major Versions (X.0.0)**:
+  - Not currently in use until we get to 1.0.
+- **Minor Versions (X.Y.0)**:
+  - Add new features in a backward-compatible manner.
+  - Introduce breaking changes after deprecation.
+- **Patch Versions (X.Y.Z)**:
+  - Introduce deprecations changes with announcements and migration paths.
+  - Include bug fixes, security updates, and performance improvements without changing functionality or APIs.
+
+### 2. Deprecation Policy
+- **Deprecation Announcement**:
+  - Any feature slated for removal will be marked as deprecated and announced in release notes.
+  - Deprecated features will include clear warnings in logs and CLI outputs.
+  - Deprecation notices will be included [Pepr Slack](https://kubernetes.slack.com/archives/c06dgh40ucb).
+- **Grace Period**:
+  - Deprecated features will remain functional for at least **one release** before removal.
+
+### 3. Test Coverage
+- Maintain backward compatibility by validating changes against existing test suites for previously supported versions.
+- Include regression tests to ensure deprecated features continue to function as expected until their removal.
```

Comment:
Curious to see how we actually implement this, does it justify an issue in the backlog so we can centralize discussion and work around this topic later?

Reply:
I think it could either go [here](https://github.com/defenseunicorns/pepr/issues/1096) if if that would be too messy it could be a [discussion](https://github.com/defenseunicorns/pepr/discussions)

Reply:
I like the idea of a new discussion around how to do compatibility testing. I'll start a discussion there since it sounds distinct from the mechanics of how we make releases with `semantic-release` or some similar tool.

Reply:
See #1716